src/app/admin/layout.tsx

```diff
@@ -2,23 +2,39 @@ import Navbar from "@/components/Navbar";
 import { container } from "@/shared/tw";
 import { Metadata } from "next";
 import { PropsWithChildren } from "react";
+import AuthProvider from "@/contexts/AuthProvider";
+import { getServerSession } from "next-auth";
+import { authOptions } from "@/shared/auth";
+import { redirect } from "next/navigation";
+import { headers } from "next/headers";
 
 export const metadata: Metadata = {
   title: "PodCodar Admin",
 };
 
-export default function RootLayout({ children }: PropsWithChildren) {
+export default async function RootLayout({ children }: PropsWithChildren) {
+  const session = await getServerSession(authOptions);
+
+  if (!session) {
+    const headerObj = headers();
+    const origin = headerObj.get("x-url") ?? "";
+
+    return redirect(`/login?callbackUrl=${origin}`);
```

Comment:
In future, let's consider move these hard-coded strings to a `constants.ts` file

Reply:
Ok, added this task to the next steps